apps/cms/src/pages/api/provider-instances.ts

```diff
@@ -0,0 +1,117 @@
+import { createProtectedHandler, NextProtectedApiHandler } from "@saleor/app-sdk/handlers/next";
+
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { saleorApp } from "../../../saleor-app";
+import { CMSSchemaProviderInstances, SingleProviderSchema } from "../../lib/cms/config";
+import { Setting } from "../../lib/cms/utils";
+import { createClient } from "../../lib/graphql";
+import { createSettingsManager } from "../../lib/metadata";
+import { generateUniqueId } from "../../modules/cms/utils";
+
+export type SettingsUpdateApiRequest = SingleProviderSchema;
+
+export interface ProviderInstancesApiResponse {
+  success: boolean;
+  data?: CMSSchemaProviderInstances | SingleProviderSchema;
+}
+
+// todo: implement
+// const obfuscateSecret = (secret: string) => {
+//   return "*".repeat(secret.length - 4) + secret.substring(secret.length - 4);
+// };
+
+const handler: NextProtectedApiHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse<ProviderInstancesApiResponse>,
+  context
+) => {
+  const { authData } = context;
+
+  const client = createClient(authData.saleorApiUrl, async () => ({
+    token: authData.token,
+  }));
+
+  const settingsManager = createSettingsManager(client);
+
+  if (req.method === "GET") {
+    const settingsManagerValue = await settingsManager.get("providerInstances");
```

Comment:
I would consider parsing the `settingsManagerValue` with `zod` to make sure we are receiving what we expect from the settingsManager.